airbyte-cdk/python/unit_tests/destinations/test_destination.py

```diff
@@ -0,0 +1,234 @@
+
+
+def _record(stream: str, data: Dict[str, Any]) -> AirbyteRecordMessage:
+    return AirbyteRecordMessage(stream=stream, data=data, emitted_at=0)
+
+
+def _spec(schema: Dict[str, Any]) -> ConnectorSpecification:
+    return ConnectorSpecification(connectionSpecification=schema)
+
+
+def write_file(path: PathLike, content: Union[str, Mapping]):
+    content = json.dumps(content) if isinstance(content, Mapping) else content
+    with open(path, 'w') as f:
+        f.write(content)
+
+
+def _wrapped(
+        msg: Union[AirbyteRecordMessage, AirbyteStateMessage, AirbyteCatalog, ConnectorSpecification, AirbyteConnectionStatus]
+) -> AirbyteMessage:
+    if isinstance(msg, AirbyteRecordMessage):
+        return AirbyteMessage(type=Type.RECORD, record=msg)
+    elif isinstance(msg, AirbyteStateMessage):
+        return AirbyteMessage(type=Type.STATE, state=msg)
+    elif isinstance(msg, AirbyteCatalog):
+        return AirbyteMessage(type=Type.CATALOG, catalog=msg)
+    elif isinstance(msg, AirbyteConnectionStatus):
+        return AirbyteMessage(type=Type.CONNECTION_STATUS, connectionStatus=msg)
+    elif isinstance(msg, ConnectorSpecification):
+        return AirbyteMessage(type=Type.SPEC, spec=msg)
+    else:
+        raise Exception(f"Invalid Airbyte Message: {msg}")
+
+
+class OrderedIterableMatcher(Iterable):
+    """
+    A class whose purpose is to verify equality of one iterable object against another
+    in an ordered fashion
+    """
+
+    def attempt_consume(self, iterator):
+        try:
+            return next(iterator)
+        except StopIteration:
+            return None
+
+    def __iter__(self):
+        return iter(self.iterable)
+
+    def __init__(self, iterable: Iterable):
+        self.iterable = iterable
+
+    def __eq__(self, other):
+        if not isinstance(other, Iterable):
+            return False
+
+        other_iter = other if isinstance(other, Iterator) else iter(other)
+        self_iter = self if isinstance(self, Iterator) else iter(self)
+        self_next = other_next = 'not none'
+        while self_next is not None and other_next is not None:
+            other_next = self.attempt_consume(other_iter)
+            self_next = self.attempt_consume(self_iter)
+            if self_next != other_next:
+                return False
+
+        return True
+
+
+class TestRun:
+    def test_run_spec(self, mocker, destination: Destination):
+        args = {'command': 'spec'}
+        parsed_args = argparse.Namespace(**args)
+
+        expected_spec = ConnectorSpecification(connectionSpecification={'json_schema': {'prop': 'value'}})
+        mocker.patch.object(
+            destination,
+            'spec',
+            return_value=expected_spec,
+            autospec=True
+        )
+
+        spec_message = next(iter(destination.run_cmd(parsed_args)))
+
+        # Mypy doesn't understand magicmock so it thinks spec doesn't have call_count attr
+        assert destination.spec.call_count == 1  # type: ignore
```

Comment:
```suggestion
        destination.spec.assert_called_once()  # type: ignore
```

Reply:
np, done, but is there a benefit to this?